oslabs-beta/DockerLocal
# Leave one out

## Description: 
- Added functionality for the composed file to only contain the check marked repositories from the client
## Changes I Made:
In `cloneRepo.sh` & `findDockerfiles.sh`
-  Changed 'myRepos' to 'myProjects' (myRepos folder changed to myProjects)

In `ProjectPage.tsx `
- For the `composeFile` function, the checked repositories are added to the body in the `POST` request to /docker

In `dockerController.ts`
- In `dockerController.createDockerCompose` 
→ Inside the file building loop - after taking the checked repos out of res.locals, a conditional only allows those repos to be used in the compose file

In `dockerRoute.ts`
- Added `authController.saveUserInfoAndRepos` to beginning of middleware chain
- With fs, fileAsJSON reads the compose file and saves it as a string
- A payload variable containing the compose file path and the compose file as a string - then sent to the client

## How to Test:
 - With a set of already cloned repos (in a project) in local storage, the compose button will send the correct JSON data to the server and a compose file should be made with the correct/checked repos only.

## Issues: 
- The compose file is sent a string, which may be difficult to parse/display on the front end
